design.md

```diff
@@ -0,0 +1,94 @@
+# y2c
+
+## Description
+
+Automatically import events from Github repos into Google Calendar.
+
+## Overview
+
+Github Action that reads event data from specific YAML files (`y2c.yml`) in a GitHub repo and calls the Google Calendar API to create/update events. The action is only triggered when master is pushed and the files containing event data change. Event attendees can either be individual's email addresses or a Google Group email address.
+
+
+## Not in scope
+
+* Managing members of the Google Groups. If members need to be added to the Google Group, this will need to happen via the Google Groups UI.
+* Creating/Updating events on groups.io
+* Using Groups from groups.io.
+
+## Format of event files (`y2c.yml`)
+
+```yaml
+id: 'event-id'
+organizer: 'dan@linuxfoundation.com' # email address of the organizer
+calendarId: 'calendar-id'           # Calendar Identifier
+title: 'Event title'
+description: 'Event Description'
+location: 'zoom-url'                # Are we using the Zoom URL as location?
+start:
+  date: '2020-02-10'                # Event start date
+  time: '9:00am'                    # Event start time
+  timeZone: 'America/New_York'
+end:
+  date: '2020-02-10'                # Event end date
+  time: '11:00am'                   # Event end time
+  timeZone: 'America/New_York'
+attendees:
+  - email: joe@example.com          # Guest email
+    name: 'Joe User'                # Guest name (optional)
+  - email: jane@example.com
+    name: 'Jane User'
+reminders:
+  - method: 'email'                 # Send email reminder 10 minutes before event
+    minutes: 10
+  - method: 'popup'                 # Show popup reminder 1 hour before event
+    minutes: 60
+```
+
+#### Notes
+* The user will have to specify an event ID, so changes to the `y2c.yml` file trigger an update on Google Calendar, instead of creating a new event. The user will have to remember not to change the eventId once set, otherwise a new event would be created on Google Calendar.
+* The user will have to specify the Calendar Identifier where the event will be created. The organizer must have access to that calendar.
+* Not sure we care about setting attendee names.
+* Not sure we want to allow users to configure the event's reminders or we want to set the same policy for all events.
```

Comment:
We should encourage sticking to Google Calendar defaults.

Reply:
OK. Should we not allow changing them?

Reply:
Let's leave it out for now but please create an issue to re-evaluate.